src/smt/smt_engine.cpp

```diff
@@ -231,6 +231,12 @@ void SmtEngine::finishInit()
     d_asserts->setProofGenerator(pppg);
     // enable it in the SmtSolver
     d_smtSolver->setProofNodeManager(pnm);
+    // if proofs and unsat cores, proofs are used solely for unsat core
+    // production, so we don't generate proofs in the theory engine
+    if (options::unsatCoresNew())
```

Comment:
I think we should make this an argument to SmtSolver finishInit and not have a separate `set` method.